presto-spi/src/main/java/com/facebook/presto/spi/plan/Assignments.java

```diff
@@ -176,14 +176,17 @@ public int hashCode()
     public static class Builder
     {
         private final Map<VariableReferenceExpression, RowExpression> assignments = new LinkedHashMap<>();
+        private boolean built;
```

Comment:
Why preventing building new assignments?

Reply:
I also don't think this is necessary. You can totally add some assignments, build it to an `UnmodifiableList`, and add more, and build it to another one. I don't see why there's problem in using the builder that way.

Reply:
`UnmodifiableMap` or `UnmodifiableList` is not as strict as `ImmutableMap` /`ImmutableList`, they do not prevent the original map/list being modified.

So if somebody build Assignments through builder then continued to add new items into the builder, it will accidentally modify the content of the assignment that has been built.

Another way is of course always make a copy when we build Assignments. But a boolean checks cost less than copying (especially when assignment can contain thousands of elements in some queries).

Reply:
Wait, so `UnmodifiableList` is not unmodifiable?

Reply:
Basically, if you
```
List a = new ArrayList();
List b = new UnmodifiableList(a);
```
You cannot modify `b` but still you can change `a` and the values in `b` will change. 
`Unbomifiable` does not prevent underlying datastructure change by other method.

Reply:
@hellium01 : As I comment https://github.com/prestodb/presto/pull/13045/commits/ec23ffe5ae6e811b28ec9e638aa11e241aba63e5#r302308058, what required is to make an immutable copy.